ingest.py

```diff
@@ -55,37 +64,61 @@ def load_single_document(file_path: str) -> Document:
     raise ValueError(f"Unsupported file extension '{ext}'")
 
 
-def load_documents(source_dir: str) -> List[Document]:
-    # Loads all documents from source documents directory
+def load_documents(source_dir: str, ignored_files: List[str] = []) -> List[Document]:
+    """
+    Loads all documents from the source documents directory, ignoring specified files
+    """
     all_files = []
     for ext in LOADER_MAPPING:
         all_files.extend(
             glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
         )
-    return [load_single_document(file_path) for file_path in all_files]
+    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    return [load_single_document(file_path) for file_path in filtered_files]
 
-
-def main():
-    # Load environment variables
-    persist_directory = os.environ.get('PERSIST_DIRECTORY')
-    source_directory = os.environ.get('SOURCE_DIRECTORY', 'source_documents')
-    embeddings_model_name = os.environ.get('EMBEDDINGS_MODEL_NAME')
-
-    # Load documents and split in chunks
+def process_documents(ignored_files: List[str] = []) -> List[Document]:
+    """
+    Load documents and split in chunks
+    """
     print(f"Loading documents from {source_directory}")
-    chunk_size = 500
-    chunk_overlap = 50
-    documents = load_documents(source_directory)
+    documents = load_documents(source_directory, ignored_files)
+    if not documents:
+        print("No new documents to load")
+        exit(0)
+    print(f"Loaded {len(documents)} new documents from {source_directory}")
     text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)
     texts = text_splitter.split_documents(documents)
-    print(f"Loaded {len(documents)} documents from {source_directory}")
-    print(f"Split into {len(texts)} chunks of text (max. {chunk_size} characters each)")
+    print(f"Split into {len(texts)} chunks of text (max. 500 tokens each)")
+    return texts
+
+def does_vectorstore_exist(persist_directory: str) -> bool:
+    """
+    Checks if vectorstore exists
+    """
+    if os.path.exists(os.path.join(persist_directory, 'index')):
+         if os.path.exists(os.path.join(persist_directory, 'chroma-collections.parquet')) and os.path.exists(os.path.join(persist_directory, 'chroma-embeddings.parquet')):
+            list_index_files = glob.glob(os.path.join(persist_directory, 'index/*.bin'))
+            list_index_files += glob.glob(os.path.join(persist_directory, 'index/*.pkl'))
+            if len(list_index_files) == 4:
```

Comment:
This implementation is quite fragile, as @PulpCattel pointed out. It may break with a new version of Chroma.

Reply:
What would be your suggeson for foolproofing the check for an existing index?

Reply:
@PulpCattel suggested checking if persist_directory is empty or not, but in my experience when ingest fails the directory contains 2 documents... so I can't think of a much better solution. Maybe using > 3 instead of == 4 makes it a little more flexible. But I don't love that solution either, so I'd be happy to go with this for the moment

Reply:
Yeah I think for now this can be a workable compromise.
If we keep this, a couple of small fixes:
* The second `if` from the top is over-indented.
* The two `path.join()` could become `path.join(persist_directory, 'index', '*.bin')`, i.e. split `'index/*.bin'` and `'index/*.pkl'`